glean-core/src/lib.rs

```diff
@@ -229,6 +229,29 @@ fn setup_state(state: State) {
     }
 }
 
+/// A global singleton that stores listener callbacks registered with Glean
+/// to receive event recording notifications.
+static EVENT_LISTENERS: OnceCell<Mutex<HashMap<String, Box<dyn GleanEventListener>>>> =
+    OnceCell::new();
+
+fn event_listeners() -> &'static Mutex<HashMap<String, Box<dyn GleanEventListener>>> {
+    if EVENT_LISTENERS.get().is_none() && EVENT_LISTENERS.set(Mutex::new(HashMap::new())).is_err() {
+        log::error!("Event Listeners object is already set, this probably happened concurrently");
+    }
+
+    EVENT_LISTENERS.get().unwrap()
```

Comment:
Is `get_or_init` not suitable here? `once_cell` has [a decent example](https://docs.rs/once_cell/latest/once_cell/#lazy-initialized-global-data).

Reply:
Ahh, nifty! You get used to the patterns you see and just keep repeating them. This seems to do the trick nicely.